keepalived-vip/utils.go

```diff
@@ -0,0 +1,219 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"errors"
+	"net"
+	"os"
+	"regexp"
+	"sort"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/unversioned"
+	k8sexec "k8s.io/kubernetes/pkg/util/exec"
+	"k8s.io/kubernetes/pkg/util/node"
+	"k8s.io/kubernetes/pkg/util/sysctl"
+)
+
+const (
+	vethRegex = "^veth.*"
+)
+
+var (
+	invalidIfaces = []string{"lo", "docker0", "flannel.1"}
+)
+
+type nodeInfo struct {
+	iface   string
+	ip      string
+	netmask int
+}
+
+// getNodeInfo returns information of the node where the pod is running
+func getNodeInfo(nodes []string) (*nodeInfo, error) {
+	ip, err := myIP(nodes)
+	if err != nil {
+		return &nodeInfo{}, err
+	}
+
+	return &nodeInfo{
+		iface:   interfaceByIP(ip),
+		ip:      ip,
+		netmask: maskForLocalIP(ip),
+	}, nil
+}
+
+// myIP returns
+func myIP(nodes []string) (string, error) {
+	var err error
+	for _, iface := range netInterfaces() {
+		ip, _, err := ipByInterface(iface.Name)
+		if err == nil && stringSlice(nodes).pos(ip) != -1 {
+			return ip, nil
+		}
+	}
+
+	glog.Errorf("error getting local IP: %v", err)
+	return "0.0.0.0", err
+}
+
+// netInterfaces returns a slice containing the local network interfaces
+// excluding lo, docker0, flannel.1 and veth interfaces.
```

Comment:
what happens if you don't ignore these interfaces, rather, what happens if i have rkt.0 or lxcbr0 or something?